linkerd/app/inbound/src/policy/store.rs

```diff
@@ -31,51 +27,46 @@ struct PortHasher(u16);
 
 // === impl Store ===
 
-impl Store {
-    fn mk_default(default: DefaultPolicy) -> Option<(Tx, Rx)> {
-        match default {
-            DefaultPolicy::Deny => None,
-            DefaultPolicy::Allow(sp) => Some(watch::channel(sp)),
-        }
-    }
-
-    pub(crate) fn fixed(
-        default: impl Into<DefaultPolicy>,
+impl<S> Store<S> {
+    pub(crate) fn spawn_fixed(
+        default: DefaultPolicy,
         ports: impl IntoIterator<Item = (u16, ServerPolicy)>,
-    ) -> (Self, Option<Tx>) {
-        let rxs = ports
-            .into_iter()
-            .map(|(p, s)| {
+    ) -> Self {
+        let cache = {
+            let rxs = ports.into_iter().map(|(p, s)| {
                 // When using a fixed policy, we don't need to watch for changes. It's
                 // safe to discard the sender, as the receiver will continue to let us
                 // borrow/clone each fixed policy.
                 let (_, rx) = watch::channel(s);
                 (p, rx)
-            })
-            .collect();
-
-        let (default_tx, default) = match Self::mk_default(default.into()) {
-            Some((tx, rx)) => (Some(tx), Some(rx)),
-            None => (None, None),
+            });
+            Cache::with_permanent_from_iter(Duration::MAX, rxs)
```

Comment:
hmm, i'm not sure about this...i feel like we should still pass in the idle timeout in the fixed policy case. we won't spawn a bunch of background tasks in the case where `discover` is `None`, but we'll still insert a bunch of clones of the default policy into the cache which will never go away. this means that something like a port scanner enumerating all ports will blow up the cache to 65535 entries, and it'll stay there forever...